cpp/examples/sbml_to_memilio.cpp

```diff
@@ -0,0 +1,921 @@
+        << std::endl;
+    parameters << "namespace mio\n{" << std::endl;
+    parameters << "namespace " << lowercase_name << "\n{" << std::endl;
+
+    std::string parameterset_initializer = "using ParametersBase = ParameterSet<";
+
+    if (number_parameters == 0) {
+        parameters << "template <typename FP = ScalarType>\nstruct DummyParam {\n    using Type = double;\n    static "
+                      "Type get_default()\n    {\n        return 0.0;\n    }\n    static std::string name()\n    {\n   "
+                      "     return \"DummyParameter\";\n    }\n};\n"
+                   << std::endl;
+        parameterset_initializer = parameterset_initializer + "DummyParam<FP>, ";
+    }
+    else {
+        for (size_t i = 0; i < number_parameters; i++) {
+            Parameter param  = *(Parameter*)model->getListOfParameters()->get(i);
+            double value     = param.getValue();
+            std::string name = param.getId();
+            parameters << "template <typename FP = ScalarType>" << std::endl;
+            parameters << "struct " << name << " {\n    using Type = double;\n  static Type get_default()\n{"
+                       << std::endl;
+            parameters << "return " << value << ";\n}" << std::endl;
+            parameters << "static std::string name()\n{\n   return \"" << name << "\";\n}\n};\n" << std::endl;
+            parameterset_initializer = parameterset_initializer + name + "<FP>, ";
+        }
+    }
+    parameterset_initializer.pop_back();
+    parameterset_initializer.pop_back();
+    parameters << "template <typename FP = ScalarType>" << std::endl;
+    parameters << parameterset_initializer << ">;\n " << std::endl;
+    parameters << "template <typename FP = ScalarType>\nclass Parameters : public ParametersBase<FP>\n{\npublic:\n    "
+                  "Parameters()\n        : ParametersBase<FP>()\n    {\n    }\n\npublic:\n    template <class "
+                  "IOContext>\n    static IOResult<Parameters> deserialize(IOContext& io)\n    {\n        "
+                  "BOOST_OUTCOME_TRY(auto&& base, ParametersBase<FP>::deserialize(io));\n        return "
+                  "success(Parameters(std::move(base)));\n    }\n\n};\n}\n}\n#endif\n"
+               << std::endl;
+
+    parameters.close();
+
+    return true;
+}
+
+/*
+    @brief Create the model.cpp file
+
+    @param filename The name of the file to be processed
+
+    Creates the generic model.cpp file. The file location is generated using the filename.
+*/
+bool create_model_cpp(const char* filename)
+{
+
+    std::string lowercase_name = boost::to_lower_copy<std::string>(get_filename(filename));
+
+    std::ofstream model_cpp;
+    model_cpp.open(lowercase_name + "/model.cpp", std::ios::out);
+
+    model_cpp << "#include \"" << lowercase_name << "/model.h\"\n\nnamespace mio\n{" << std::endl;
+    model_cpp << "namespace " << lowercase_name << "\n{\n\n}\n}" << std::endl;
+    model_cpp.close();
```

Comment:
Another note on file operations, since you use the file (via fstream) in a very limited scope, you don't *have to* call close manually, the destructor can take care of that. I am not sure what is best practice here, so I leave it up to you whether you want to keep the close, or remove them.